code/game/objects/effects/turf_fire.dm

```diff
@@ -0,0 +1,183 @@
+#define TURF_FIRE_REQUIRED_TEMP (T0C+10)
+#define TURF_FIRE_TEMP_BASE (T0C+100)
+#define TURF_FIRE_POWER_LOSS_ON_LOW_TEMP 7
+#define TURF_FIRE_TEMP_INCREMENT_PER_POWER 3
+#define TURF_FIRE_VOLUME 150
+#define TURF_FIRE_MAX_POWER 50
+
+#define TURF_FIRE_ENERGY_PER_BURNED_OXY_MOL 12000
+#define TURF_FIRE_BURN_RATE_BASE 0.12
+#define TURF_FIRE_BURN_RATE_PER_POWER 0.02
+#define TURF_FIRE_BURN_CARBON_DIOXIDE_MULTIPLIER 0.75
+
+#define TURF_FIRE_STATE_SMALL 1
+#define TURF_FIRE_STATE_MEDIUM 2
+#define TURF_FIRE_STATE_LARGE 3
+
+/obj/effect/abstract/turf_fire
+	icon = 'icons/effects/turf_fire.dmi'
+	icon_state = "fire_small"
+	layer = BELOW_OPEN_DOOR_LAYER
+	anchored = TRUE
+	move_resist = INFINITY
+	light_range = 1.5
+	light_power = 1.5
+	light_color = LIGHT_COLOR_FIRE
+	mouse_opacity = FALSE
+	/// How much power have we got. This is treated like fuel, be it flamethrower liquid or any random thing you could come up with
+	var/fire_power = 20
+	/// Is it magical, if it is then it wont interact with atmos, and it will not loose power by itself. Mainly for adminbus events or mapping
+	var/magical = FALSE
```

Comment:
i would suggest splitting this into atmos_interact and passive_loss (probably a better anme?)

Reply:
done